ledger/src/blockstore.rs

```diff
@@ -148,6 +148,52 @@ pub struct BlockstoreSignals {
     pub completed_slots_receiver: CompletedSlotsReceiver,
 }
 
+#[derive(Default)]
+pub struct TurbineFecSetStats(HashMap<Slot, HashMap</*fec_set_index*/ u32, /*count*/ usize>>);
+
+impl TurbineFecSetStats {
+    pub fn inc_index_count(&mut self, slot: Slot, fec_set_index: u32) {
+        *self
+            .0
+            .entry(slot)
+            .or_default()
+            .entry(fec_set_index)
+            .or_default() += 1;
+    }
+
+    pub fn get_min_index_count(&self, slot: &Slot) -> (usize, usize) {
+        let m = self.0.get(slot);
+        if m.is_none() {
+            return (0, 0);
+        }
+        let m = m.unwrap();
+
+        let last_idx = m.iter().map(|(idx, _)| *idx).max();
+        if last_idx.is_none() {
+            return (0, 0);
+        }
+        let last_idx = last_idx.unwrap();
+
+        let min_fec_set_cnt = m
+            .iter()
+            .filter(|(idx, _)| **idx != last_idx)
+            .map(|(_, cnt)| *cnt)
+            .min()
+            .unwrap_or(0);
+        let last_idx_cnt = *m.get(&last_idx).unwrap_or(&0);
```

Comment:
What is the reasoning for excluding last batch from the others?
Batches are always the same size:
https://github.com/solana-labs/solana/blob/15357480e/ledger/src/shred.rs#L962-L968

Reply:
The number of data shreds in the last batch could be less than `MAX_DATA_SHREDS_PER_FEC_BLOCK` so it could skew the min batch size computation.

Reply:
I don't recall any discussions about excluding data shreds. To my understanding, for any erasure batch we are counting how many shreds, either data or code, were received from turbine.

The probabilities and simulations we have discussed on discord were agnostic to if the shreds are data or code.